File: iOS/SurveyMaster/Model/Question Types/RangeQuestion.swift
//
//  RangeQuestion.swift
//  SurveyMaster
//
//  Created by Ammar Al-Helali on 6/25/19.
//  Copyright © 2019 Ammar Al-Helali. All rights reserved.
//

import Foundation
public class RangeQuestion : Question {
  
    var content = Content()
     struct Content {
        var min : Int?
        var max : Int?
        var minLabel : Int?
        var maxLabel : Int?
        var minDefaultValue : Int?
        var maxDefaultValue : Int?
        var step : Int?
    }
//    override init(_ title : String , _ description : String) {
//        super.init(title, description)
//    }
    
    required init(from decoder: Decoder) throws {
        fatalError("init(from:) has not been implemented")
    }
    func setInfo(cell: Any) {
        
    }
    func setJSON (cell: Any){
        
    }
    
}

File: server/models/questions/rangeQuestion.js
const Joi = require('joi')
const Question = require('./question')
const types = require('../types')
const { Questions } = require('../validationSchemas')

class RangeQuestion extends Question {
    constructor(props) {
        super(props)
        this.type = types.QUESTION_RANGE
        this.content = {
            min: props.content.min || 0,
            max: props.content.max || 5,
            minLabel: props.content.minLabel || 'minimum',
            maxLabel: props.content.maxLabel || 'maximum',
            minDefaultValue: props.content.minDefaultValue || props.content.min,
            maxDefaultValue: props.content.maxDefaultValue || props.content.max,
            step: props.content.step || 1 // if -1 -> there is no step
        }
    }

    static validate(rangeQuestion) {
        const result = Joi.validate(rangeQuestion, Questions.rangeQuestionSchema)
        return result
    }
}

module.exports = RangeQuestion